src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchive.cs

```diff
@@ -211,6 +212,20 @@ public ZipArchive(Stream stream, ZipArchiveMode mode, bool leaveOpen, Encoding?
             }
         }
 
+        /// <summary>
+        /// Gets or sets the optional archive comment.
+        /// </summary>
+        /// <remarks>
+        /// The comment encoding is determined by the <c>entryNameEncoding</c> parameter of the <see cref="ZipArchive(Stream,ZipArchiveMode,bool,Encoding?)"/> constructor.
+        /// If the comment byte length is larger than <see cref="ushort.MaxValue"/>, it will be truncated when disposing the archive.
+        /// </remarks>
+        [AllowNull]
+        public string Comment
+        {
+            get => (EntryNameAndCommentEncoding ?? Encoding.UTF8).GetString(_archiveComment);
```

Comment:
Why do we assume UTF8 if EntryNameAndCommentEncoding is null? Meanwhile, `GetEncodedTruncatedBytesFromString` calls `GetEncoding`, which takes the encoding from analyzing the string.

Reply:
I first need to read the one passed by the user. If that one is empty, I need UTF8 to be the fallback always, since it can detect both unicode and ASCII characters.

Reply:
> I need UTF8 to be the fallback always, since it can detect both unicode and ASCII characters.

Why we don't always return UTF8 on `GetEncoding` then?

Reply:
EntryNameAndCommentEncoding should've gotten its value earlier from what the user passed, or what we initially set if it was null.